src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2450,7 +2450,21 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
                     }
 
                     var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
-                    if (incompleteMember.ContainsDiagnostics)
+
+                    var containsErrorDiagnostic = false;
+                    foreach (var child in incompleteMember.EnumerateNodes())
+                    {
+                        foreach (var diagnostic in child.GetDiagnostics())
+                        {
+                            if (diagnostic.Severity == DiagnosticSeverity.Error && !diagnostic.IsWarningAsError)
```

Comment:
So that turning on "warnings as errors" doesn't change the diagnostics that get printed.